### DISCUSSION TITLE
Add support for hasOneThrough in Fieldsets and group

### DISCUSSION THREAD

**Discussion by jhhazelaar** [filamentphp/filament#3615] [Ideas]
Add support for hasOneThrough relationship in fieldsets relationship.

We get the type error:
```
Filament\Forms\Components\Fieldset::getRelationship(): Return value must be of type Illuminate\Database\Eloquent\Relations\BelongsTo|Illuminate\Database\Eloquent\Relations\HasOne|Illuminate\Database\Eloquent\Relations\MorphOne|null, Illuminate\Database\Eloquent\Relations\HasOneThrough returned
``` 

It is quite the same as a HasOne relationship right?

**Comment by danharrin** [comment]
Maybe you can have multiple nested `relationship()` methods to simulate a through relationship

```php
Group::make()
    ->schema([
        Fieldset::make()
            ->schema([
                //
            ])
            ->relationship('secondrelation')
    ])
    ->relationship('firstrelation')
```

**Reply by jhhazelaar** [reply]
I thought I tested this and didn't work, I thought. But it actually works! Thanks for the tip.

**Reply by jhhazelaar** [reply]
I've tested it with two Fieldsets, maybe that's the difference.

**Comment by jhhazelaar** [comment]
My reply above was a bit to soon. With this schema you can edit the fields, but can't add them. So when creating a record with nested relationships, it isn't working.

Is this a bug or feature request?

**Reply by danharrin** [reply]
Probably a feature request, we should work out how to support HasOneThrough on fieldsets without using 2

**Reply by yagogak** [reply]
That work pretty well like this !
